src/plugin/timezone/index.js

```diff
@@ -107,11 +110,21 @@ export default (o, c, d) => {
     return ins
   }
 
-  proto.offsetName = function (type) {
+  proto.offsetName = function (type = 'short') {
     // type: short(default) / long
     const zone = this.$x.$timezone || d.tz.guess()
+    const cachedOffsetName = this.$x.$offsetName || {}
+    const cacheKey = `${zone}-${type}}`
+    if (cachedOffsetName[cacheKey]) {
+      return cachedOffsetName[cacheKey]
+    }
     const result = makeFormatParts(this.valueOf(), zone, { timeZoneName: type }).find(m => m.type.toLowerCase() === 'timezonename')
-    return result && result.value
+    if (type === 'short') {
```

Comment:
Thanks for putting this PR together, I found it really helpful! 🎉  

There's a small issue where the `value` for Istanbul is 'GMT+03:00' so the returned abbreviation becomes `G`.

Maybe you could add a check to see if there are any digits in the value and then return the value instead of abbreviating it?